src/App.tsx

```diff
@@ -132,38 +105,45 @@ function App() {
                 )
               }
             </HStack>
-            {state.publishState == "Setup" ||
-            state.publishState == "Connecting" ? (
+            {publisherState == "ready" ||
+              publisherState == "connecting" ? (
               <Button
-                isLoading={state.publishState == "Connecting"}
-                onClick={() => {
-                  dispatch({ type: "join" });
-                  setTimeout(() => {
-                    dispatch({ type: "joined" });
-                  }, 1000);
-                }}
+                  isLoading={publisherState == "connecting"}
+                  onClick={(async () => {
+                    // This needs to actually launch preview mode and not start streaming
+                    await navigator.mediaDevices.getUserMedia({ audio: true, video: true }).then((mediaDevice) => {
+                      const broadcastOptions: BroadcastOptions = {
+                        mediaStream: mediaDevice
+                      }
+                      startStreaming(broadcastOptions);
+                    });
+                  })}
               >
                 Join
               </Button>
             ) : undefined}
             {
               /** TODO: create a streaming control component */
-              state.publishState == "Ready" && (
-                <Button onClick={() => dispatch({ type: "goLive" })}>
+              publisherState == "ready" && (
+                <Button onClick={() => {
```

Comment:
we don't need 'GoLive' button anymore, right?

Reply:
Not right now - but it'll be needed when we have that video/audio preview screen. So I've just left it as is for the moment.